### DISCUSSION TITLE
Podman / Fedora / SELinux

### DISCUSSION THREAD

**Discussion by aclater** [automatic-ripping-machine/automatic-ripping-machine#1254] [General]
A few years ago [I rewrote the install script](https://github.com/automatic-ripping-machine/automatic-ripping-machine/discussions/557) to work on Fedora Server 36 - It looks like @PriamX picked that up and got it working on fedora-37. Generally, the install scripts were pretty simple and easy to pick through and figure out what was missing and why. This never really got taken up or adopted (likely my fault for not submitting a PR), and I see the decision was later made to focus on just [Docker containers and Ubuntu 20.04 LTS](https://github.com/automatic-ripping-machine/automatic-ripping-machine/discussions/698).

Yesterday I started playing around with putting arm into a [Quadlet](https://github.com/containers/quadlet) so that arm can be run as a user under Podman/SystemD. I got it working, but it was pretty ugly - I had to disable SELinux to get it to start, and then once it was started, it now works fine with SELinux enabled. I believe these issues can be resolved with some nominal effort.

For example - looking over docker-setup.sh, it only took me about ten or fifteen minutes to quickly pull together a script which accomplished the same outcomes on my Fedora 41 box. It would be nice if the script would identify the OS it is running under and adapt accordingly. The differences are nominal.

I'm trying to figure out if there is enough interest in pursuing this further - and if there is any interest by @shitwolfymakes and the rest of the devs to make the installation a little more OS agnostic - the differences between fedora and ubuntu are really pretty insignificant and with [Podman](https://github.com/containers/podman) moving into CNCF - I expect usage and uptake of it to continue to increase.

Anyways - if this is a discussion better suited for another place or time, let me know. I think the opportunity exists for a container which can operate in either environment - but it may take some effort.

**Comment by microtechno9000** [comment]
ARM 3.x is being worked on at the moment, moving to docker-compose structure.

Thats the main focus for the small dev team.

You are welcome to play around and raise a PR if there are improvements you would like to see. For example add in the fedora manual build script and updating the wiki, or testing out podman and making changes to suit.

**Reply by aclater** [reply]
be gentle - its my first pull req https://github.com/automatic-ripping-machine/automatic-ripping-machine/pull/1255
